Add InlineModelAdmin.has_add_permission definition

JoaRiski opened the pull request:
Add a proper definition for the InlineModelAdmin's has_add_permission method.

Fixes #469

**NOTE: This change is entirely untested, so take extra care when reviewing.** I'm unsure if the type check works in stub inheritance the same way it does in actual code, but if yes it could be this change will hit the same "incompatible with supertype" error as mentioned in #469

JoaRiski committed:
Add InlineModelAdmin.has_add_permission definition

Add a proper definition for the InlineModelAdmin's has_add_permission method.

Fixes #469

blueyed reviewed (approved):
LGTM.

The difference with InlineModelAdmin is that the parent object gets passed in there.

https://github.com/django/django/blob/742d6bc8dbc3868201895933e1c40c36f46462c7/django/contrib/admin/options.py#L2087
https://github.com/django/django/blob/742d6bc8dbc3868201895933e1c40c36f46462c7/django/contrib/admin/options.py#L2170-L2177

Can you rebase it to re-trigger CI?

sobolevn committed:
Merge branch 'master' into patch-1

JoaRiski committed:
Improve ModelAdmin.has_add_permission declaration

Improve how `has_add_permission` is declared by removing it from `BaseModelAdmin` and adding it directly to `ModelAdmin`. This is required as `InlineModelAdmin` has a different signature, see #469

sobolevn reviewed (commented):
Or maybe we can use `# type: ignore` on `Inline`'s version of this method.

I think, that it will be more accurate. In this case, subclasses of `BaseModelAdmin` would still have `has_add_permission` method.
